forest/src/daemon.rs

```diff
@@ -223,6 +225,18 @@ pub(super) async fn start(config: Config, detached: bool) {
         config
     };
 
+    // Terminate if no snapshot is provided or DB isn't recent enough
```

Comment:
There is already a lot of complexity in the `daemon.rs` file. We can do all this auto download snapshot logic in `main.rs`. We don't need any service of the daemon for this (RPC, libp2p, etc).

So basically move all the logic before testing the `detach` flag:

ie
```
            match cmd {
                Some(command) => {
                    task::block_on(subcommand::process(command, cfg));
                }
                None => {
                    task::block_on(prompt_and_fetch_snapshot(&mut cfg));
                    if opts.detach {
                        ...
```

Reply:
We do, we need the `chain_store` to check if the DB is out of date, which is one of the criteria for offering to download a snapshot